src/python/pants/base/workunit.py

```diff
@@ -119,6 +119,7 @@ def __init__(self, run_info_dir, parent, name, labels=None, cmd='', log_config=N
     # In seconds since the epoch. Doubles, to account for fractional seconds.
     self.start_time = 0
     self.end_time = 0
```

Comment:
Because these properties are public, it would be good to defend against people accessing them directly rather than consuming `def duration`. Perhaps you could convert them into `@property`s that fail if `_synthetic_duration` is set? Or convert to `@property`s and then deprecate the properties?